### PR TITLE
Gets Rid of Both Sleeps In Transit Tubes, Replaces Them With Timers

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
Every tube that the transit pod moves through causes two sleeps, each at most 2 deciseconds. now it uses a timer on the delay that is always > 0 and on the delay thats usually 0 it checks whether to make a timer. 

## Why It's Good For The Game
sleeps bad

## Changelog
:cl:
code: transit tubes are less sleepy now
/:cl:

### CODE REVIEW

**Comment by Kylerace** [comment]
ah oops forgot i did the effects thing in this pr uno momento

**Comment by CRITAWAKETS** [comment]
You left some commented out code, no point in keeping it in!

Also, kinda wish this was wording on the timer subsystem and using delta_time, but this is still infinitely better than sleep() so, good job on this.

**Comment by Kylerace** [comment]
> 
> 
> You left some commented out code, no point in keeping it in!
> 
> Also, kinda wish this was wording on the timer subsystem and using delta_time, but this is still infinitely better than sleep() so, good job on this.

tried making it a process() proc, couldnt figure it out unless i wanted to get rid of one of the delays and make the other one locked at 2 deciseconds (or make a new process child just for transit tubes but idk if thats icky or not)

File: code/game/objects/structures/transit_tubes/transit_tube_pod.dm [hunk 1]
```diff
@@ -104,52 +113,53 @@
...
+				Move(get_step(loc, dir), dir, DELAY_TO_GLIDE_SIZE(exit_delay)) // Allow collisions when leaving the tubes.
+				return
 
-		if(current_tube == null)
+			enter_delay = current_tube.enter_delay(src, next_dir)
+			if(enter_delay > 0)
+				addtimer(CALLBACK(src, .proc/move_animation, MOVE_ANIMATION_STAGE_TWO), enter_delay)
+				return
+			else
+				return .(MOVE_ANIMATION_STAGE_TWO)
```
Comment (Mothblocks) [inline code/game/objects/structures/transit_tubes/transit_tube_pod.dm:151]: 1. Don't use `.` like this, it's weirdly obfuscating.
2. I don't think we need to go through the overhead of recursively calling this proc. I say make this do `stage = MOVE_ANIMATION_STAGE_TWO`, remove the switch, and make it two separate if statements.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/transit_tubes/transit_tube_pod.dm (hunk 1 above)
Comment (Kylerace) [reply code/game/objects/structures/transit_tubes/transit_tube_pod.dm]: done

**Pull request merged**